hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+import { useState, useEffect } from 'react';
+import { Dispatch, SetStateAction } from 'react';
+
```

Comment:
refactor (nit): use a type alias to communicate better what the type is, e.g.

```suggestion
type LocalStorageReturnType<T> = [T, Dispatch<SetStateAction<T>>]

```

and use it everywhere `[T, Dispatch<SetStateAction<T>>]` is.